lib/src/node/value/calculation.dart

```diff
@@ -0,0 +1,144 @@
+// Copyright 2023 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:collection/collection.dart';
+import 'package:node_interop/js.dart';
+import 'package:sass/src/node/immutable.dart';
+import 'package:sass/src/node/utils.dart';
+import 'package:sass/src/util/nullable.dart';
+
+import '../../value.dart';
+import '../reflection.dart';
+
+/// Check that [arg] is a valid argument to a calculation function.
+void isCalculationValue(Object arg, {bool checkUnquoted = true}) {
+  if (arg is! SassNumber &&
+      arg is! SassString &&
+      arg is! SassCalculation &&
+      arg is! CalculationOperation &&
+      arg is! CalculationInterpolation) {
+    jsThrow(JsError('Argument must be one of '
+        'SassNumber, SassString, SassCalculation, CalculationOperation, '
+        'CalculationInterpolation'));
+  }
+  if (checkUnquoted && arg is SassString && arg.hasQuotes) {
+    jsThrow(JsError('Argument must be unquoted SassString'));
+  }
+}
+
+/// The JavaScript `SassCalculation` class.
+final JSClass calculationClass = () {
+  var jsClass =
+      createJSClass('sass.SassCalculation', (Object self, [Object? _]) {
+    jsThrow(JsError("new sass.SassCalculation() isn't allowed"));
+  });
+
+  jsClass.defineStaticMethods({
+    'calc': (Object argument) {
+      isCalculationValue(argument);
+      return SassCalculation.unsimplified('calc', [argument]);
+    },
+    'min': (Object arguments) {
+      var argList = jsToDartList(arguments).cast<Object>();
+      argList.forEach(isCalculationValue);
+      return SassCalculation.unsimplified('min', argList);
+    },
+    'max': (Object arguments) {
+      var argList = jsToDartList(arguments).cast<Object>();
+      argList.forEach(isCalculationValue);
+      return SassCalculation.unsimplified('max', argList);
+    },
+    'clamp': (Object min, [Object? value, Object? max]) {
+      if (value == null && max == null) {
+        String minString;
+        if (min is SassString) {
+          minString = min.text;
+        } else if (min is CalculationInterpolation) {
+          minString = min.value;
+        } else {
+          jsThrow(JsError(
+              '`value` and `max` are both undefined and `min` is not a SassString or CalculationInterpolation'));
+        }
+        final values = minString
+            .split(',')
+            .map((v) => num.tryParse(v.trim()).andThen((v) => SassNumber(v)))
```

Comment:
@nex I'm trying to parse the `min` argument of `clamp()` as a comma-separated list of `SassNumber`s. This approach works well with something like `1, 2, 3`, but fails if the numbers contain units: `1px, 2rem, 3ch`. I can extract the units for `SassNumber` with a regex, but wanted to check with you if there's something in the codebase that already does this

Reply:
@jerivas I think you mean @nex3 👍🏻